"""Tests pour vÃ©rifier la compatibilitÃ©
des packages utilisÃ©s par Airflow

Contexte: au 2025-01-15 via PR1200 on est confrontÃ© au
problÃ¨me suivant:
    ğŸŸ¢ les tests dags_unit_tests/ fonctionnent
    ğŸŸ¢ la CI Github fonctionne
    ğŸ”´ Mais les DAGs Ã©chouent car pandas 2.2.3 casse la
    compatibilitÃ© avec notre Engine SQLAlchemy

D'oÃ¹ des tests python, indÃ©pendents des DAGs Airflow et la CI
pour vÃ©rifier les compatibilitÃ©s de base
"""

import pandas as pd
from sqlalchemy import create_engine


def test_pandas_read_sql_table():
    """Au 2025-01-15 ce test fonctionne en pandas 2.1.4 mais
    Ã©choue en 2.2.3 avec l'erreur suivante:
    AttributeError: 'Engine' object has no attribute 'cursor'

    TODO: lorsque ce test Ã©choue Ã  l'avenir Ã  cause d'une upgrade pandas,
    penser Ã :
     - mettre Ã  jour ce teste pour qu'il fonctionne
     - mettre Ã  jour les codes DAGs pour qu'ils fonctionnent Ã©galement
    """
    print(f"Pandas version: {pd.__version__}")
    engine = create_engine("sqlite:///:memory:")
    engine.execute("CREATE TABLE my_table (id INT, name TEXT)")
    pd.read_sql_table("my_table", engine)
